subworkflows/local/utils_nfcore_measeq_pipeline/main.nf

```diff
@@ -61,13 +57,28 @@ workflow PIPELINE_INITIALISATION {
         nextflow_cli_args
     )
 
+    //
+    // Summarize and Validate Params
+    //
+    if (validate_params) {
+        validateParameters()
+    }
+    log.info paramsSummaryLog(workflow)
+
     //
     // Create channel from input file provided through params.input
     //
     Channel
-        .fromList(samplesheetToList(params.input, "${projectDir}/assets/schema_input.json"))
+        .fromSamplesheet("input")
         .map {
             meta, fastq_1, fastq_2 ->
+                // Meta ID assignment
+                if (!meta.id) {
+                    meta.id = meta.irida_id
+                } else {
+                    meta.id = meta.id.replaceAll(/[^A-Za-z0-9_.\-]/, '_')
+                }
```

Comment:
Could you also add a `while` loop here similar to https://github.com/phac-nml/speciesabundance/blob/main/workflows/speciesabundance.nf#L86-L88

That bit of code is present in order to handle the situation where multiple samples have the same name. In that case, we'd differentiate them by appending the IRIDA Next identifier at the end. That is, if two samples had name `SampleA` and one had IRIDA Next ID `ID2`, you'd end up with two samples where the `meta.id` would be:

* `SampleA`
* `SampleA_ID2`